docs/ShowInDialogButton.md

```diff
@@ -5,129 +5,247 @@ title: "ShowInDialogButton"
+    <ShowInDialogButton fullWidth maxWidth="md">
+-       <SimpleShowLayout>
++       <TabbedShowLayout>
++           <TabbedShowLayout.Tab label="Main">
+                <TextField source="first_name" fullWidth />
+                <TextField source="last_name" fullWidth />
+                <DateField source="dob" label="born" fullWidth />
+                <SelectField source="sex" choices={sexChoices} fullWidth />
++           </TabbedShowLayout.Tab>
+-       </SimpleShowLayout>
++       </TabbedShowLayout>
+    <ShowInDialogButton fullWidth maxWidth="md">
+);
+```
+
+You can also pass a React element as child, to build a custom layout. Check [Building a custom Show Layout](./ShowTutorial.md#building-a-custom-layout) for more details.
+
+## `ButtonProps`
+
+The `ButtonProps` prop allows you to pass props to the MUI `<Button>` component. For instance, to change the color of the button, you can use the `color` prop:
+
+{% raw %}
+```jsx
+const ShowButton = () => (
+  <ShowInDialogButton ButtonProps={{ color: 'primary' }}>
+      <SimpleShowLayout>
+          ...
+      </SimpleShowLayout>
+  </ShowInDialogButton>
+);
+```
+{% endraw %}
+
+## `emptyWhileLoading`
+
+By default, `<ShowInDialogButton>` renders its child component even before the `dataProvider.getOne()` call returns. If you use `<SimpleShowLayout>` or `<TabbedShowLayout>`, this isn't a problem as these components only render when the record has been fetched. But if you use a custom child component that expects the record context to be defined, your component will throw an error.
+
+To avoid this, set the `emptyWhileLoading` prop to `true`:
+
+```jsx
+const ShowButton = () => (
+  <ShowInDialogButton emptyWhileLoading>
+      ...
+  </ShowInDialogButton>
+);
+```
+
+## `icon`
+
+The `icon` prop allows you to pass an icon to the button. It can be a MUI icon component, or a custom icon component.
+
+```jsx
+import { Edit } from '@mui/icons-material';
+
+const ShowButton = () => (
+  <ShowInDialogButton icon={<Edit />}>
+      ...
+  </ShowInDialogButton>
+);
+```
+
+## `id`
+
+The `id` prop allows you to pass the record id to the `<ShowInDialogButton>` component. If not provided, it will be deduced from the record context.
+
+This is useful to link to a related record. For instance, the following button lets you show the author of a book:
+
+```jsx
+const ShowAuthorButton = () => {
+  const book = useRecordContext();
+  return (
+    <ShowInDialogButton resource="authors" id={book.author_id}>
+        ...
+    </ShowInDialogButton>
+  );
+};
+```
+
+## `inline`
+
+By default, `<ShowInDialogButton>` renders a `<Button>` component. If you want to display only an `<IconButton>`, set the `inline` prop to `true`:
+
+```jsx
+const ShowButton = () => (
+  <ShowInDialogButton inline>
+      ...
+  </ShowInDialogButton>
+);
+```
+
+## `label`
+
+The `label` prop allows you to pass a custom label to the button, instead of the default ("Show"). It can be a string, or a React element.
```

Comment:
```suggestion
The `label` prop allows you to pass a custom label to the button, instead of the default ("Show"). It can be a string, a I18N value, or a React element.
```